client/packages/system/src/ReasonOption/Components/ReasonOptionsSearchInput.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FC } from 'react';
+import {
+  Autocomplete,
+  BasicTextInput,
+  Box,
+  defaultOptionMapper,
+  getDefaultOptionRenderer,
+  ReasonOptionNode,
+  ReasonOptionNodeType,
+} from '@openmsupply-client/common';
+import { reasonOptions } from '../api';
+
+interface ReasonOptionsSearchInputProps {
+  value?: ReasonOptionNode | null;
+  width?: number | string;
+  onChange: (reasonOption: ReasonOptionNode | null) => void;
+  autoFocus?: boolean;
+  type: ReasonOptionNodeType;
+  isError?: boolean;
+  isDisabled?: boolean;
+}
+
+export const ReasonOptionsSearchInput: FC<ReasonOptionsSearchInputProps> = ({
+  value,
+  width,
+  onChange,
+  autoFocus = false,
+  type,
+  isError,
+  isDisabled,
+}) => {
+  const { data, isLoading } = reasonOptions.document.listAllActive();
+
+  const reasonFilter = (reason: ReasonOptionNode) => {
+    switch (type) {
+      case ReasonOptionNodeType.PositiveInventoryAdjustment:
+        return reason.type === ReasonOptionNodeType.PositiveInventoryAdjustment;
+      case ReasonOptionNodeType.NegativeInventoryAdjustment:
+        return reason.type === ReasonOptionNodeType.NegativeInventoryAdjustment;
+      case ReasonOptionNodeType.RequisitionLineVariance:
+        return reason.type === ReasonOptionNodeType.RequisitionLineVariance;
+      case ReasonOptionNodeType.ReturnReason:
+        return reason.type === ReasonOptionNodeType.ReturnReason;
+      default:
+        return ReasonOptionNodeType.PositiveInventoryAdjustment;
```

Comment:
This default is a little funky, your `reasonFilter` function should probably only return a boolean right?